test/testutil/testdata.go

```diff
@@ -14,6 +15,12 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+var valueBundleBufferPool = sync.Pool{
```

Comment:
We might want to use `&sync.Pool`, because the documentation states:

> A Pool must not be copied after first use.

I don't see anything in your code that would cause a copy, but if we use a reference, it's much less likely somebody will make a mistake in the future.